File: app/src/main/java/com/airfreshener/telegram_sms/qrcode_show_activity.java
package com.airfreshener.telegram_sms;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.widget.ImageView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.github.sumimakito.awesomeqrcode.AwesomeQrRenderer;
import com.google.gson.Gson;
import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;

public class qrcode_show_activity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_qrcode);
        Context context = getApplicationContext();
        SharedPreferences sharedPreferences = context.getSharedPreferences("data", MODE_PRIVATE);
        config_list config = new config_list();
        config.bot_token = sharedPreferences.getString("bot_token", "");
        config.chat_id = sharedPreferences.getString("chat_id", "");
        config.trusted_phone_number = sharedPreferences.getString("trusted_phone_number", "");
        config.fallback_sms = sharedPreferences.getBoolean("fallback_sms", false);
        config.chat_command = sharedPreferences.getBoolean("chat_command", false);
        config.battery_monitoring_switch = sharedPreferences.getBoolean("battery_monitoring_switch", false);
        config.charger_status = sharedPreferences.getBoolean("charger_status", false);
        config.verification_code = sharedPreferences.getBoolean("verification_code", false);
        config.privacy_mode = sharedPreferences.getBoolean("privacy_mode", false);
        ImageView qr_image_imageview = findViewById(R.id.qr_imageview);
        qr_image_imageview.setImageBitmap(new AwesomeQrRenderer().genQRcodeBitmap(new Gson().toJson(config), ErrorCorrectionLevel.H, 1024, 1024));
    }


    private static class config_list {
        String bot_token = "";
        String chat_id = "";
        String trusted_phone_number = "";
        boolean fallback_sms = false;
        boolean chat_command = false;
        boolean battery_monitoring_switch = false;
        boolean charger_status = false;
        boolean verification_code = false;
        boolean privacy_mode = false;
    }
}

File: app/src/main/java/com/airfreshener/telegram_sms/QrCodeShowActivity.kt
package com.airfreshener.telegram_sms

import android.os.Bundle
import android.widget.ImageView
import androidx.appcompat.app.AppCompatActivity
import com.github.sumimakito.awesomeqrcode.AwesomeQrRenderer
import com.google.gson.Gson
import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel

class QrCodeShowActivity : AppCompatActivity() {

    private val prefsRepository by lazy { (application as TelegramSmsApp).prefsRepository }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_qrcode)
        val settings = prefsRepository.getSettings()
        val config = ConfigList(
            bot_token = settings.botToken,
            chat_id = settings.chatId,
            trusted_phone_number = settings.trustedPhoneNumber,
            fallback_sms = settings.isFallbackSms,
            chat_command = settings.isChatCommand,
            battery_monitoring_switch = settings.isBatteryMonitoring,
            charger_status = settings.isChargerStatus,
            verification_code = settings.isVerificationCode,
            privacy_mode = settings.isPrivacyMode,
        )
        val qrImageImageview = findViewById<ImageView>(R.id.qr_imageview)
        qrImageImageview.setImageBitmap(
            AwesomeQrRenderer().genQRcodeBitmap(Gson().toJson(config),
                ErrorCorrectionLevel.H,
                1024,
                1024
            )
        )
    }

    @Suppress("unused", "PropertyName")
    private class ConfigList(
        val bot_token: String?,
        val chat_id: String?,
        val trusted_phone_number: String?,
        val fallback_sms: Boolean,
        val chat_command: Boolean,
        val battery_monitoring_switch: Boolean,
        val charger_status: Boolean,
        val verification_code: Boolean,
        val privacy_mode: Boolean,
    )
}
